src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -797,21 +917,47 @@ bool DirectoryService::RunConsensusOnDSBlockWhenDSPrimary() {
   }
 
   auto sortedDSPoWSolns = SortPoWSoln(allDSPoWs);
-  auto sortedPoWSolns = SortPoWSoln(allPoWs, true);
 
-  map<PubKey, Peer> powDSWinners;
+  std::map<PubKey, Peer> powDSWinners;
+  std::vector<PubKey> removeDSNodePubkeys;
   MapOfPubKeyPoW dsWinnerPoWs;
   uint32_t version = DSBLOCK_VERSION;
   uint8_t dsDifficulty = 0;
   uint8_t difficulty = 0;
   uint64_t blockNum = 0;
   BlockHash prevHash;
 
-  unsigned int numOfProposedDSMembers = ComputeDSBlockParameters(
-      sortedDSPoWSolns, sortedPoWSolns, powDSWinners, dsWinnerPoWs,
-      dsDifficulty, difficulty, blockNum, prevHash);
+  // Determine the DS PoW winners.
+  unsigned int numOfProposedDSMembers =
+      ComputeDSBlockParameters(sortedDSPoWSolns, powDSWinners, dsWinnerPoWs,
+                               dsDifficulty, difficulty, blockNum, prevHash);
+
+  // Determine the losers from the performance.
+  unsigned int numByzantine = 0;
+  if (m_mediator.m_currentEpochNum >= ENABLE_NEW_FEATURE_DS_NUM) {
```

Comment:
Change to m_mediator.m_dsBlockChain.GetLastBlock().GetHeader().GetBlockNum() >=
      ENABLE_NEW_FEATURE_DS_NUM